pkg/ring/ring.go

```diff
@@ -461,6 +462,100 @@ func (r *Ring) Subring(key uint32, n int) ReadRing {
 	return sub
 }
 
+// ShuffleShard returns a subring for the provided identifier (eg. a tenant ID)
+// and size (number of instances). The size is expected to be a multiple of the
+// number of zones and the returned subring will contain the same number of
+// instances per zone as far as there are enough registered instances in the ring.
+//
+// The algorithm used to build the subring is a shuffle sharder based on probabilistic
```

Comment:
What do you mean by "probabilistic hashing"?

Reply:
Uniqueness is not guaranteed, but it offsers a low collision probability. What we're using here is a [probabilistic data structure](https://towardsdatascience.com/hashes-power-probabilistic-data-structures-d1398d1335c6).